src/main/java/co/cask/hydrator/plugin/spark/dynamic/SparkCompilers.java

```diff
@@ -214,4 +218,30 @@ private static void copyPathAndAdd(FileSystem fs, Path from, File dir, Collectio
   private SparkCompilers() {
     // no-op
   }
+
+  /**
+   * Recursively delete a directory.
+   */
+  public static void deleteDir(@Nullable File dir, Logger logger) {
```

Comment:
Better define a static logger in this class rather than having it passes in. Otherwise it's hard to relate the logger name (classname) to where the log happens.